pkg/git/git.go

```diff
@@ -80,8 +80,165 @@ func (h *stiGit) Clone(source, target string) error {
 	}
 	go pipeToLog(outReader, glog.Info)
 	go pipeToLog(errReader, glog.Error)
-	return h.runner.RunWithOptions(opts, "git", "clone", "--quiet", "--recursive", source, target)
+	//previously, we were doing the git clone with --quiet, but with the piping of the wrapping cmd stdout/stderr
+	// to glog on separate threads, we were causing a deadlock processing stdout/stderr channels across 4 goroutines.
+	// a SIGQUIT produced goroutine thread dump that shows the deadlock is included  after this method for historical
+	// purposes.
+	// Since it seems contradictory to run with --quiet and also pipe git output to glog, the workaround to the apparent
+	// go issue is the remove the --quiet option.  I also saved the core file produced by the SIGQUIT.
+	// a discussion on whether this is frequent enough or bad enough to open a Go issue on github could be warranted.
```

Comment:
Yeah, I think if still desired a verbal conversation could flush out the diagnosis.  But I just had a bit of an additional break through ...

I was going to note when I dove into your review responses tomorrow that I originally tried not launching the go routines that called pipeToLog to glog since --quiet was specified, and that did NOT do the trick, and so I fell back onto removing --quiet.

But a lightbulb popped on about 15 minutes ago and I realized that attempt was not quite right.  Couldn't wait until Monday to try it and report ;-) ....It is not just the go routines, but the setting of the CommandOpts Stdout and Stderr that is at issue here when --quiet is specified (and hence no output is sent to those pipes).  I mentioned in the big temporary comment I left for this discussion that was at contention too.

I just tried it - when I change the clone method to just

	opts := util.CommandOpts{
	}
	return h.runner.RunWithOptions(opts, "git", "clone", "--quiet", "--recursive", source, target)

Then the git clone of my fork of simple-openshift-sinatra-sti runs perfectly.

So yeah, if you don't want any output from git clone to show up ... i.e. -- quiet, then the whole creation of pipes and redirection of logging make no sense.  

I also tried an experiment with just specifying stderr (in case --quiet does not supress that), and that appears to be working too (though my very first one may have hung, but I may have aborted it too quickly).  I did also retry both stdout and stderr and it still failed.

I'll push either the no pipes and stderr only pipe change (if I continue to see consistent good results) with the other responses to your comments on Monday.